lib/engine/game/g_1893/step/buy_train.rb

```diff
@@ -1,12 +1,31 @@
 # frozen_string_literal: true
 
 require_relative '../../../step/buy_train'
+require_relative '../buy_eva_from_other_player'
 
 module Engine
   module Game
     module G1893
       module Step
         class BuyTrain < Engine::Step::BuyTrain
+          def actions(entity)
+            possible_actions = super
+
+            return [] if possible_actions.empty?
+
+            possible_actions << 'buy_company'
+          end
+
+          include BuyEvaFromOtherPlayer
```

Comment:
doesn't seem necessary

Reply:
I'll revisit my solution, and update the PR.